pkg/security/probe/model_ebpf.go

```diff
@@ -37,9 +38,27 @@ func NewEBPFModel(probe *EBPFProbe) *model.Model {
 	}
 }
 
-// NewEBPFEvent returns a new event
-func NewEBPFEvent(fh *EBPFFieldHandlers) *model.Event {
+func newEBPFEvent(fh *EBPFFieldHandlers) *model.Event {
 	event := model.NewFakeEvent()
 	event.FieldHandlers = fh
 	return event
 }
+
+// newEBPFEventFromPCE returns a new event from a process cache entry
+func newEBPFEventFromPCE(entry *model.ProcessCacheEntry, fh *EBPFFieldHandlers) *model.Event {
+	eventType := model.ExecEventType
+	if !entry.IsExec {
+		eventType = model.ForkEventType
+	}
```

Comment:
Are we sure we want to send fork events too?

Reply:
or add a filter looking at subscriber event types to return earl if nobody wants it?

Reply:
Added something to in the probe_ebpf.go side to not generate event if there is no fork subscriber